src/plugin/microphone.js

```diff
@@ -13,309 +13,338 @@
  */
 
 /**
- * Microphone plugin definition factory
+ * Visualise microphone input in a wavesurfer instance.
  *
- * @param  {MicrophonePluginParams} params parameters use to initialise the plugin
- * @return {PluginDefinition} an object representing the plugin
+ * @implements {PluginClass}
+ * @extends {Observer}
+ * @example
+ * // es6
+ * import MicrophonePlugin from 'wavesurfer.js/dist/plugin/wavesurfer.microphone.js';
```

Comment:
while it's technically correct, I would leave out the whole path and just import from 'wavesurfer.microphone.js'.